stl/inc/vector

```diff
@@ -440,6 +440,193 @@ constexpr _Ty* _Unfancy_maybe_null(_Ty* _Ptr) noexcept { // do nothing for plain
     return _Ptr;
 }
 
+#ifdef __SANITIZE_ADDRESS__
+extern "C" void __sanitizer_annotate_contiguous_container(
+    const void* _First, const void* _End, const void* _Old_last, const void* _New_last) noexcept;
+
+template <class _Vec, class = void>
+_INLINE_VAR constexpr bool _Has_minimum_allocation_alignment =
+    disjunction_v<bool_constant<alignof(typename _Vec::value_type) >= 8>,
+        is_same<typename _Vec::allocator_type, allocator<typename _Vec::value_type>>>;
+
+template <class _Vec>
+_INLINE_VAR constexpr bool
+    _Has_minimum_allocation_alignment<_Vec, void_t<decltype(_Vec::allocator_type::_Minimum_allocation_alignment)>> =
+        _Vec::allocator_type::_Minimum_allocation_alignment >= 8;
+
+template <class _Ty, bool = _Has_minimum_allocation_alignment<_Ty>>
+class _NODISCARD _Asan_annotater {
+private:
+    using _Alloc      = typename _Ty::allocator_type;
+    using _Pointer    = typename _Ty::pointer;
+    using _Size_type  = typename _Ty::size_type;
+    using _Value_type = typename _Ty::value_type;
+
+public:
+    explicit constexpr _Asan_annotater(const _Ty& _Target) noexcept : _Target_vec(_Target), _Old_last(nullptr) {}
+
+    _CONSTEXPR20 _Asan_annotater(const _Ty& _Target, const _Size_type _New_size)
+        : _Target_vec(_Target), _Old_last(nullptr) {
+#if _HAS_CXX20
+        if (_STD is_constant_evaluated()) {
+            return;
+        }
+#endif // _HAS_CXX20
+
+        const auto& _My_data     = _Target_vec._Mypair._Myval2;
+        const _Pointer& _Myfirst = _My_data._Myfirst;
+        const _Pointer& _Mylast  = _My_data._Mylast;
+        const _Pointer& _Myend   = _My_data._Myend;
+
+        if (_New_size > static_cast<_Size_type>(_Myend - _Myfirst)) {
+            // New size is larger than current capacity; assume we're allocating a new block.
+            return;
+        }
+
+        // New size is less than current capacity; assume we're reusing the same block.
+        _Old_last = _Unfancy_maybe_null(_Mylast);
+
+        if (_New_size > static_cast<_Size_type>(_Mylast - _Myfirst)) {
+            const auto _Raw_first = _Unfancy(_Myfirst);
+            __sanitizer_annotate_contiguous_container(_Raw_first, _Unfancy(_Myend), _Old_last, _Raw_first + _New_size);
+        }
+    }
+
+    _CONSTEXPR20 ~_Asan_annotater() {
+#if _HAS_CXX20
+        if (_STD is_constant_evaluated()) {
+            return;
+        }
+#endif // _HAS_CXX20
+
+        const auto& _My_data     = _Target_vec._Mypair._Myval2;
+        const _Pointer& _Myfirst = _My_data._Myfirst;
+        const _Pointer& _Mylast  = _My_data._Mylast;
+        const _Pointer& _Myend   = _My_data._Myend;
+
+        if (!_Old_last) {
+            const _Raw_end = _Unfancy(_Myend);

```

Comment:
This line is malformed, causing test failures:

```
D:\build\out\inc\vector(508): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
```

Occurs below.

Reply:
`const void*` would work, and let us get rid of the icky cast on 514.